pages/management/applications.md

```diff
@@ -4,76 +4,121 @@ title: Applications
+When you flash this image onto your device, the device will automatically appear in your application dashboard—no manual intervention is required. You can use this image file for multiple devices, and resin.io will create a unique ID and name for each one.
 
 
-### Deploying Your Code to an Application
+### Deploy code to an application
 
-The key thing to know about any application from your perspective as a developer is it's git endpoint - this is visible in the applications list on the [applications dashboard](http://dashboard.resin.io) and also in the top-right hand corner of each individual application dashboard.
+Each application has an associated **git** endpoint, which follows the syntax `<USERNAME>@git.resin.io:<USERNAME>/<APPNAME>.git`. In the top-right corner of your application page, you'll find the command to add this endpoint as a **git** remote:
 
-To configure a git repo to be able to push code to resin, you need to add a [git remote](http://gitref.org/remotes/) - simply click the button to the right of the git endpoint to copy the command to the clipboard and run it in the folder where your local git repo is located.
+<img src="/img/common/app/remote-repo.png" width="80%">
 
-Alternatively, simply run `git remote add resin [git endpoint]`, and you're done. From then on in you can simply run `git push resin master` to push your master branch to your devices.
+When you are ready to deploy your code, navigate to your project directory, copy and paste the command from your application page, and push your master branch to the `resin` remote:
+
+```shell 
+$ cd FirstApp
+$ git remote add resin <USERNAME>@git.resin.io:<USERNAME>/<APPNAME>.git
+$ git push resin master
+```
+__Note:__ If you have made local changes to your code, make sure to `git add` and `git commit` as necessary before pushing to the remote.
+
+Once you push the code to your resin.io application, a **Docker** container will be built on our servers and downloaded to any online devices.
 
 For more details on deployment, check out our [deployment guide](/deployment/deployment).
 
-## Application Actions
+## Device filters
+
+As the number of devices in your application grows, the device list will become increasingly busy. Filters provide a convenient way to quickly find specific devices based on shared characteristics. 
+
+### Add and update filters
+
+To add a filter, click the *Add filter* button near the top-left corner of your application page. You'll be presented with a window to configure your filter:
+
+<img src="/img/common/app/add-filter.png" width="80%">
+
+The first dropdown lists the device characteristics you can filter on. These characteristics correspond to the columns in your device list. The second dropdown contains comparison operators appropriate to the device characteristic, such as `is`, `is before`, `contains`, and `matches RegEx`. The third field is where you specify a value to filter on. Click *Add filter* to apply your configuration.
```

Comment:
this is true for now, but just a heads up that when we add tags, you will also be able to filter based on tags. I'm sure we will also create new docs for tags, and can update the filters docs as needed when the time comes.